atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/DbKvs.java

```diff
@@ -568,58 +573,91 @@ private ColumnSelection getColumnSelectionForRange(RangeRequest range) {
         return rows;
     }
 
+    public void setGetRangeOfTsMaxBatch(long newValue) {
+        getRangeOfTsMaxBatch = newValue;
+    }
+
+    public long getGetRangeOfTsMaxBatch() {
+        return getRangeOfTsMaxBatch;
+    }
+
+    /**
+     * @param tableRef the name of the table to read from.
+     * @param rangeRequest the range to load.
+     * @param timestamp the maximum timestamp to load.
+     *
+     * @return Each RowResult that is returned contains at most getRangeOfTsMaxBatch associated timestamps. It is
+     * guaranteed that all timestamps for a row will be returned in successive RowResults, sorted in ascending order
+     * by column name and timestamp. In addition, the last column and timestamp of each RowResult will be repeated to
+     * ensure sweep does not miss timestamps.
```

Comment:
This sounds like a hack. Can we do better on the sweep side, e.g. by remembering what the latest timestamp we've seen so far was?